test_requirements.txt

```diff
@@ -10,6 +10,7 @@ pickle5; python_version == '3.7' and platform_python_implementation != 'PyPy'
 cffi
 # For testing types. Notes on the restrictions:
 # - Mypy relies on C API features not present in PyPy
-# - There is no point in installing typing_extensions without mypy
 mypy==0.910; platform_python_implementation != "PyPy"
-typing_extensions==3.10.0.0; platform_python_implementation != "PyPy"
+
+# TODO: Remove once 3.10.0.2 has been released (xref python/typing#865)
+typing_extensions==3.10.0.0; python_version == '3.10'
```

Comment:
The new 3.10.0.1 release is currently broken for python 3.10 (https://github.com/python/typing/issues/865).
Pin it to 3.10.0.0 to prevent mypy from using this broken version.